unity/README.md

```diff
@@ -0,0 +1,17 @@
+# .NET Runtime - Unity Details
+
+This is Unity's fork of the .NET Runtime repository.
+
+The difference between this fork and the upstream repository should be as small as possible, with all of the differences specific to Unity. Our goal is to upstream as many changes made here as possible.
+
+## Pulling changes from upstream
+
+The [`main`](https://github.com/Unity-Technologies/runtime/tree/main) branch should always be in sync with the upstream [dotnet/runtime](https://github.com/dotnet/runtime) repository. There is a job in Unity's internal CI which runs weekly to pull the latest code from the upstream repo to this branch. The job will create a pull request to merge these changes to the [`unity-main`](https://github.cds.internal.unity3d.com/unity/il2cpp/pull/3602) branch.
+
+## Pushing changes to upstream
+
+When a pull request is open against this fork, we should determine if the changes in that pull request should be pushed upstream (most should). Ideally, pull request should be organized so that all changes in a given pull request can be directly applied upstream. Any changes specific to the Unity fork should be done in a separate pull request.
+
+Assuming the branch with changes to upstream is named `great-new-feature` then a new branch of [`main`](https://github.com/Unity-Technologies/runtime/tree/main) named `upstream-great-new-feature` should be created. Each commit from `great-new-feature` should be cherry-picked `upstream-great-new-feature`, and then a pull request should be opened from `upstream-great-new-feature` to [`main`](https://github.com/dotnet/runtime/tree/main) in the upstream repository.
```

Comment:
I suspect we mean unity-main here and not main? And why do this at all?  The upstream merge should get this later. Or conflict? I may just be confused, which points to more clarity in this doc is needed.

Reply:
Good point, maybe this is not necessary. I thought was that we don't want to wait to merge changes in our fork for them to first land upstream, then wait a week for the sync to happen from upstream.

My idea here is that we merge the changes to `unity-main` in our fork, then create a new branch for the upstream pull request. So long as we cherry-pick the commits from `unity-main`, the sync after they land upstream should work cleanly.

Reply:
I still do not think you want `main` in this section. We should never to PR into Unity's main.  Only Unity's `unity-main`. I think that our `main` should never have any different commits from upstream.

Also consider deleting our `main`.  We do not need it.

Reply:
> We should never to PR into Unity's main.

I agree. But don't we need to make a branch of the Unity fork of `main` to allow us to open a PR to the upstream `main`?

> Also consider deleting our main. We do not need it.

I think we do need a branch to match the upstream `main`, right?